pkg/oci/oci.go

```diff
@@ -188,11 +188,11 @@ func pullIfNotExist(ctx context.Context, imageStore oras.Target, authOpts *AuthO
 		return fmt.Errorf("resolving image %q: %w", image, err)
 	}
 
-	repo, err := NewRepository(image, authOpts)
+	repo, err := NewRepository(targetImage.String(), authOpts)
```

Comment:
Do we need this change? 

If yes, should we change other places as well?

Reply:
Yes, we need it and probably other places too
I've tried with `ghcr.io/burak-ok/trace_exec` and that way it didn't add automatically the `:latest` tag, etc...

I am currently in favor of making the normalized image a type to mitigate further issues where we forgot to "normalize" the image

Reply:
There are two changes on this commit, I agree the change below has to be done, but I'm wondering about this on the `NewRepository` line. I tried the following:

on main (121d76f3fea2c87527d595eac70156cb53cbdede)

```bash
$ sudo -E ig run  ghcr.io/burak-ok/trace_exec
INFO[0000] Experimental features enabled                
Error: getting gadget info: getting gadget image: pulling image "ghcr.io/burak-ok/trace_exec": downloading to local repository: failed to resolve ghcr.io/burak-ok/trace_exec: invalid reference
```

Performed the following change

```diff
$ git diff
diff --git pkg/oci/oci.go pkg/oci/oci.go
index 934a23f2..fa0761ac 100644
--- pkg/oci/oci.go
+++ pkg/oci/oci.go
@@ -192,7 +192,7 @@ func pullIfNotExist(ctx context.Context, imageStore oras.Target, authOpts *AuthO
        if err != nil {
                return fmt.Errorf("creating remote repository: %w", err)
        }
-       _, err = oras.Copy(ctx, repo, image, imageStore, image, oras.DefaultCopyOptions)
+       _, err = oras.Copy(ctx, repo, targetImage.String(), imageStore, targetImage.String(), oras.DefaultCopyOptions)
        if err != nil {
                return fmt.Errorf("downloading to local repository: %w", err)
        }

```

And then it worked fine:

```bash
$ sudo -E ig run  ghcr.io/burak-ok/trace_exec
INFO[0000] Experimental features enabled                
RUNTIME.CONTAINERNAME                  PID                  PPID                 COMM             UID                  GID                 RETVAL   
```

So probably we only need this change?

Reply:
We need that also in other parts of it:
```bash
➭ go run --exec "sudo -E" ./cmd/ig/... image push burakokms/trace_exec
INFO[0000] Experimental features enabled                
Pushing burakokms/trace_exec...
Error: pushing gadget: copying to remote repository: Head "https://burakokms/v2/trace_exec/manifests/sha256:30507a96ca406d5453ca99b7ed8b51cfb09c1a54bb0127b56c155c8f68ba7b49": dial tcp: lookup burakokms: no such host
exit status 1

# (I didn't log in but the registry is the right one)
➭ go run --exec "sudo -E" ./cmd/ig/... image push burakokms/trace_exec
INFO[0000] Experimental features enabled                
Pushing burakokms/trace_exec...
Error: pushing gadget: copying to remote repository: HEAD "https://registry-1.docker.io/v2/burakokms/trace_exec/manifests/sha256:30507a96ca406d5453ca99b7ed8b51cfb09c1a54bb0127b56c155c8f68ba7b49": response status code 401: Unauthorized
exit status 1
```

Do we want to support this case without the full url?

Reply:
For now I removed the line in `NewRepository`.

The discussion around if we want to have and support "default registries" has to be done before changing this

Reply:
Discussion can be done in Issue https://github.com/inspektor-gadget/inspektor-gadget/issues/2190